src/main/java/org/prebid/server/bidder/adxcg/AdxcgBidder.java

```diff
@@ -46,59 +45,53 @@ public Result<List<BidderBid>> makeBids(BidderCall<BidRequest> httpCall, BidRequ
         final List<BidderError> errors = new ArrayList<>();
         try {
             final BidResponse bidResponse = mapper.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return Result.of(extractBids(bidResponse, bidRequest, errors), errors);
+            return Result.of(extractBids(bidResponse, errors), errors);
         } catch (DecodeException e) {
             return Result.withError(BidderError.badServerResponse(e.getMessage()));
         }
     }
 
-    private static List<BidderBid> extractBids(BidResponse bidResponse, BidRequest bidRequest,
-                                               List<BidderError> errors) {
+    private static List<BidderBid> extractBids(BidResponse bidResponse, List<BidderError> errors) {
         if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
             return Collections.emptyList();
         }
-        return bidsFromResponse(bidResponse, bidRequest, errors);
+        return bidsFromResponse(bidResponse, errors);
     }
 
-    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse,
-                                                    BidRequest bidRequest,
-                                                    List<BidderError> errors) {
+    private static List<BidderBid> bidsFromResponse(BidResponse bidResponse, List<BidderError> errors) {
         return bidResponse.getSeatbid().stream()
                 .filter(Objects::nonNull)
                 .map(SeatBid::getBid)
                 .filter(Objects::nonNull)
                 .flatMap(Collection::stream)
-                .map(bid -> createBidderBid(bid, bidRequest, bidResponse, errors))
+                .map(bid -> makeBid(bid, bidResponse.getCur(), errors))
                 .filter(Objects::nonNull)
                 .toList();
     }
 
-    private static BidderBid createBidderBid(Bid bid,
-                                             BidRequest bidRequest,
-                                             BidResponse bidResponse,
-                                             List<BidderError> errors) {
+    private static BidderBid makeBid(Bid bid, String currency, List<BidderError> errors) {
         try {
-            final BidType bidType = getBidType(bid, bidRequest.getImp());
-            return BidderBid.of(bid, bidType, bidResponse.getCur());
+            final BidType bidType = getBidMediaType(bid);
+            return BidderBid.of(bid, bidType, currency);
```

Comment:
imho it doesn't make much sense moving it outside since we return a bid or null depending on the bid type, so it will require additional if-statement and so on... not a big deal, if you still think it's better to do, let me know

Reply:
Try/catch adds mutch logic under the hood, for performance reasons it is much better to use plain if

But for this case, we can go with existing solution